tests/compas/data/test_copy.py

```diff
@@ -0,0 +1,21 @@
+from compas.data import Data
+
+
+class TestData(Data):
+    @property
+    def __data__(self):
+        return {}
+
+
+def test_copy():
+    data = TestData()
```

Comment:
Perhaps you need to assign a name in your test before copying the object, in order the cover Line294. Otherwise it is just `None`, right?

Reply:
the problem does not arise when a name is assigned, but when there isn't. when no name is assigned, the name attribute is not included in the serialisation info. however, (before this PR) when a copy is made the name attribute is assigned regardless. this means that the serialised info before and after copy don't match. that is what this PR fixes. the test as written, would have failed before this PR...

Reply:
ah I see the reason. I was suggesting this only because of the Codecov warning: 
<img width="493" alt="image" src="https://github.com/user-attachments/assets/f7049af7-fa5f-4b43-9bb5-a2ab755ff915">
How about doing this test both with and without name?